memory/src/main/scala/filodb.memory/format/vectors/DeltaDeltaVector.scala

```diff
@@ -253,6 +257,20 @@ object DeltaDeltaConstDataReader extends LongVectorDataReader {
       out
     }
   }
+
+  def changes(vector: BinaryVectorPtr, start: Int, end: Int): Double = {
+    require(start >= 0 && end < length(vector), s"($start, $end) is out of bounds, length=${length(vector)}")
+    var prev = Double.NaN
+    var changes = 0
+    var i = start
+    while (i <= end) {
+      val current = iterate(vector, i).next
```

Comment:
There are several problems here:
1) It's not possible for a DeltaDelta to ever return NaN, as DeltaDeltas only ever return Long integer values.
2) What is happening when you call `Double.isNan` on `current`, which is a Long, is that Long is always converted to a Double, and isNaN will always return false.  So it's wasted.
3) Calling `iterate` returns an allocation every time, which is also a waste.

I would rewrite the changes methods in the LongBinaryVectors and elsewhere to not ever compare NaN as this is not possible, and to return Int counts always.